R/utils-various.R

```diff
@@ -19,3 +19,49 @@ extend_bool <- function(value, n_match, value_name, match_name) {
     value
   }
 }
+
+#' @param x Argument passed in calling function, e.g. `int64`.
+#' @param is_missing Is `x` missing in the calling function?
+#' @param default The default of `x` in the calling function
+#' @noRd
+use_option_if_missing <- function(x, is_missing, default) {
+  nm <- deparse(substitute(x))
+  if (is_missing) {
+    x <- getOption(paste0("polars.to_r_vector_", nm), default)
+    if (!identical(x, default)) {
+      inform(
+        sprintf(
+          '`%s` is overridden by the option "%s" with %s',
+          nm,
+          paste0("polars.to_r_vector_", nm),
+          rlang:::obj_type_friendly(x)
+        ),
+      )
+    }
+  }
+  x
+}
+
+#' @param x Option value set by user.
+#' @param option_name Name of the option set by user, e.g. `"to_r_vector_int64"`.
+#' @param option_value Option value that requires checking for package presence,
+#' e.g. `"integer64"`
+#' @param pkg Name of package that must be present to use this `option_value`,
+#' e.g. `"bit64".`
+#' @noRd
+check_option_required_package <- function(x, option_name, option_value, pkg) {
+  if (x == option_value && !pkg %in% .packages()) {
+    abort(
+      paste0(
+        "package `",
+        pkg,
+        "` must be attached to use `",
+        option_name,
+        " = \"",
+        option_value,
+        "\"`."
+      ),
+      call = caller_env()
+    )
+  }
+}
```

Comment:
Move these to `polars_options.R`, as I believe this is where we would put a copy of the Python Polars functions.

Reply:
Done